### PR TITLE
Add `auth0 user import` command

### PR DESCRIPTION
**Pull request by tylernix**
### Description

This PR adds support for importing users via the `auth0 users import` command. If a template is not mentioned, it provides a dropdown that allows a user to select from 4 default user import examples. If the "Empty" template is selected, it will open a text editor to allow a person to add the users they want to import. 

a0-demo-user-import

### References

- https://auth0.com/docs/manage-users/user-migration/bulk-user-imports
- https://auth0.com/docs/manage-users/user-migration/bulk-user-import-database-schema-and-examples
- https://github.com/go-auth0/auth0/blob/master/management/job.go

### Checklist

- [x] Pass all automated checks
- [x] Add support to allow a user to open an empty user import template in an editor before importing.
- [x] Update `/docs`

### CODE REVIEW

**Review by sergiughf — CHANGES_REQUESTED** [review]
Thanks for the contribution @tylernix, solid work here. I think the feature is a good addition to the cli. 

However there are a couple of things we could do a bit better. When testing this locally and selecting any of the templates (except the empty one) we immediately import the example users from the templates, which I believe is not exactly intended. 

IMO after selecting the connection, we should keep showing the template examples (removing the empty one) and afterwards have another input where we allow users to either use the editor to add the contents or point to a file that has the desired users to import. Wdyt? 

cc: @Widcket as I'd love to hear her thoughts as well on this.

**Comment by tylernix** [comment]
@sergiughf I actually had this working the way you described, but I decided to change it since I thought the only time you would want to import your own would be on the "Empty" state, but I can see why others would want to edit the other examples before importing. The new commit always opens in editor now. I also added a "Do you want to import these users?" prompt after saving/closing the editor as an extra check.

File: internal/users/data/custom-password-hash-example.json [hunk 1]
```diff
@@ -0,0 +1,96 @@
+[
+    {
+        "email": "antoinette@contoso.com",
```
Comment (sergiughf) [inline internal/users/data/custom-password-hash-example.json:3]: I think we could keep just one option here now that the template gets opened by default and users can edit it, so they wouldn't have to remove so much in here. Wdyt?

File: internal/users/data/basic-example.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+      "email_verified": false,
+      "app_metadata": {
+          "roles": ["admin"],
+          "plan": "premium"
+      },
+      "user_metadata": {
+          "theme": "light"
+      }
+    }
+  ]
```
Comment (sergiughf) [inline internal/users/data/basic-example.json:13]: Missing new line at the end of the file 🙏🏻

File: internal/users/data/empty-example.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[
+
+]
```
Comment (sergiughf) [inline internal/users/data/empty-example.json:3]: Missing new line 🙏🏻

File: internal/users/data/mfa-factors.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
+[
+    {
+        "email": "antoinette@contoso.com",
```
Comment (sergiughf) [inline internal/users/data/mfa-factors.json:3]: Same thing as above, I think we could keep just one entry here so it's easier to edit.

File: internal/users/data/mfa-factors.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        "email": "edward@contoso.com",
+        "mfa_factors": [
+            {
+                "email": {
+                    "value": "edward@edward.biz"
+                }
+            }
+        ]
+    }
+]
```
Comment (sergiughf) [inline internal/users/data/mfa-factors.json:52]: Missing new line 🙏🏻

File: internal/users/users.go [hunk 6]
```diff
@@ -0,0 +1 @@
+package users
```
Comment (sergiughf) [inline internal/users/users.go:1]: Is there a specific reason why we have this empty file over here?

File: internal/users/data/custom-password-hash-example.json (hunk 1 above)
Comment (tylernix) [reply internal/users/data/custom-password-hash-example.json:3]: I pulled these examples directly from the user import docs. I think each one is important. 
https://auth0.com/docs/manage-users/user-migration/bulk-user-import-database-schema-and-examples#examples

File: internal/users/data/mfa-factors.json (hunk 4 above)
Comment (tylernix) [reply internal/users/data/mfa-factors.json:3]: I think each example is useful, per the user import schema docs -> https://auth0.com/docs/manage-users/user-migration/bulk-user-import-database-schema-and-examples#examples

File: internal/users/users.go (hunk 6 above)
Comment (sergiughf) [reply internal/users/users.go:1]: > Each folder in golang apparently needs to have a file with the same name
This is definitely a good heuristic to follow, however it's not something that's enforced anywhere. IMO we could remove the `users.go` file and leave the `users_embed.go` untouched.

Comment (tylernix) [reply internal/users/users.go:1]: For some reason, I thought removing `user.go` broke the compile/build so I left it, but I just deleted it, and it doesn't look like it so I'll just leave `users_embed.go`.

**Review by sergiughf — APPROVED** [review]
Solid work here @tylernix. 💪🏻

**Pull request merged**